controllers/flinkcluster_updater.go

```diff
@@ -205,6 +219,47 @@ func (updater *_ClusterStatusUpdater) deriveClusterStatus() flinkoperatorv1alpha
 			}
 	}
 
+	// (Optional) JobManager ingress.
+	var observedJmIngress = updater.observedState.jmIngress
+	if observedJmIngress != nil {
+		var state string
+		var urls []string
+		if len(observedJmIngress.Spec.TLS) > 0 {
+			for _, tls := range observedJmIngress.Spec.TLS {
+				for _, host := range tls.Hosts {
+					if host != "" {
+						urls = append(urls, "https://"+host)
+					}
+				}
+			}
+		} else {
+			for _, rule := range observedJmIngress.Spec.Rules {
+				if rule.Host != "" {
+					urls = append(urls, "http://"+rule.Host)
+				}
```

Comment:
If we make host optional, the IP address will be dynamically allocated, so here we need to use IP https://godoc.org/k8s.io/api/core/v1#LoadBalancerIngress